### PR TITLE
Fix getter type

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/ast/Types.cpp [hunk 1]
```diff
@@ -2206,7 +2206,10 @@ FunctionType::FunctionType(VariableDeclaration const& _varDecl):
 				if (auto arrayType = dynamic_cast<ArrayType const*>(member.type.get()))
 					if (!arrayType->isByteArray())
 						continue;
-				m_returnParameterTypes.push_back(member.type);
+				m_returnParameterTypes.push_back(ReferenceType::copyForLocationIfReference(
```
Comment (axic) [inline libsolidity/ast/Types.cpp]: Hm, can we have a test for this?

Comment (chriseth) [reply libsolidity/ast/Types.cpp]: It only turned up when I activated the strict decoder, not sure this is testable.

Comment (chriseth) [reply libsolidity/ast/Types.cpp]: Ah ok, right, I'll add a more direct test.

**Comment by chriseth** [comment]
@axic added test.

**Review by axic — APPROVED** [review]

**Comment by axic** [comment]
@chriseth annoyingly the gnosis git checkout is failing, restarted it once already, but no effect.

I think it should be fine to merge, all those tests are covered on circleci.

**Pull request merged**